opentracing/tracer.h

```diff
@@ -1,322 +1,39 @@
-#ifndef OPENTRACING_TRACER_H
-#define OPENTRACING_TRACER_H
+#ifndef INCLUDED_OPENTRACING_TRACER_H
+#define INCLUDED_OPENTRACING_TRACER_H
 
-#include <string>
-#include <stdint.h>
+#include <opentracing/carriers.h>
+#include <opentracing/span.h>
+#include <opentracing/spanoptions.h>
+#include <opentracing/stringref.h>
 
-namespace opentracing
-{
+namespace opentracing {
 
-class Span;
-class Writer;
-class Reader;
+class Tracer {
+  private:
+    static Tracer* s_tracer;
 
-/**
- * Options used when you start a `Span`.
- */
-struct StartSpanOptions
-{
-    /**
-     * The parent span. `NULL` if there's no parent `Span`.
-     */
-    const Span * parent;
+  public:
+    virtual SpanOptions* makeSpanOptions() = 0;
```

Comment:
why do implementations need to create their own versions of `SpanOptions`? I would have expected it to be a glorified struct, implemented directly in the opentracing lib.